docs/tutorials/earthquake_detection.ipynb

```diff
@@ -0,0 +1,704 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "executionInfo": {
+     "elapsed": 188020,
+     "status": "ok",
+     "timestamp": 1742053720038,
+     "user": {
+      "displayName": "Daniele Rege Cambrin (Darth Reca)",
+      "userId": "08456786386580134960"
+     },
+     "user_tz": -60
+    },
+    "id": "VzxO4HqUt3-A",
+    "outputId": "8ca24f21-22ed-452f-aa4c-09daf7d2951a"
+   },
+   "outputs": [],
+   "source": [
+    "embeddings = defaultdict(list)\n",
+    "labels = defaultdict(list)\n",
+    "# We iterate over the train_dataloader\n",
+    "for i, batch in tqdm(enumerate(datamodule.train_dataloader()), desc='Train'):\n",
+    "    for j, (emb, lab) in enumerate(zip(*infer(batch))):\n",
+    "        embeddings['train'].append(emb)\n",
+    "        labels['train'].append(lab)\n",
+    "# We iterate over the test_dataloader\n",
+    "for i, batch in tqdm(enumerate(datamodule.test_dataloader()), desc='Test'):\n",
+    "    for j, (emb, lab) in enumerate(zip(*infer(batch))):\n",
+    "        embeddings['test'].append(emb)\n",
+    "        labels['test'].append(lab)\n",
+    "# Now we merge the embeddings and labels into a single array\n",
+    "embeddings = {k: np.stack(v) for k, v in embeddings.items()}\n",
+    "labels = {k: np.array(v) for k, v in labels.items()}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we can fit a classical model (e.g., Random Forest) using the embeddings as features and the labels as targets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "executionInfo": {
+     "elapsed": 7180,
+     "status": "ok",
+     "timestamp": 1742053727478,
+     "user": {
+      "displayName": "Daniele Rege Cambrin (Darth Reca)",
+      "userId": "08456786386580134960"
+     },
+     "user_tz": -60
+    },
+    "id": "9q8OqFoHt3-A",
+    "outputId": "3fdba6f0-3d6a-49e5-8d07-46aacbaf04e3"
+   },
+   "outputs": [],
+   "source": [
+    "# Train a RandomForest classifier\n",
+    "clf = RandomForestClassifier(n_estimators=100, n_jobs=-1)\n",
+    "clf.fit(embeddings['train'], labels['train'])\n",
+    "# Evaluate the classifier on the test set\n",
+    "preds = clf.predict(embeddings['test'])\n",
+    "print(classification_report(labels['test'], preds))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training a deep model from scratch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "hQITTSEEt3-A"
+   },
+   "source": [
+    "The second approach requires training a deep neural network from scratch. To this end, we can use the TorchGeo's ```ClassificationTask``` and Lightning's ```Trainer``` to simplify the training.\n",
```

Comment:
All triple backticks can be replaced with single backticks. This is regular markdown, not rst.